torch/distributed/_tensor/random.py

```diff
@@ -45,7 +45,8 @@ def manual_seed(seed: int, device_mesh: DeviceMesh, tp_dim: int = 0) -> None:
     Args:
         seed (int): The desired seed.
         device_mesh (:class:`DeviceMesh`): The device mesh to set the seed.
-        tp_dim (int, optional): The mesh dimension where to apply Tensor Parallel
+        tp_dim (int, optional): The mesh dimension where to apply Tensor Parallel.
```

Comment:
I think 🤔  maybe we should remove this argument, in particular right now we are using 1-D device mesh for TP APIs instead of 2D mesh + tp_dim, so the `manual_seed(tp_mesh)` should work instead of passing a 2D DeviceMesh..

Maybe this argument can be replaced by a `rng_tracker_type` optional arg to let user specify whether to use OffsetBasedTracker or TPRNGTracker